crates/relayer/src/chain/namada/query.rs

```diff
@@ -118,27 +117,23 @@ impl NamadaChain {
     pub fn query_tx_events(&self, tx_hash: &TmHash) -> Result<Vec<IbcEventWithHeight>, Error> {
         match self.query_applied_event(tx_hash)? {
             Some(applied) => {
-                let h = applied
-                    .read_attribute::<HeightAttr>()
-                    .map_err(|_| Error::invalid_height_no_source())?;
-                let height = ICSHeight::new(self.config.id.version(), h.0)
+                let tx_response = NamadaTxResponse::try_from(applied)
+                    .expect("Could not extract tx response from a Namada event");
+                let height = ICSHeight::new(self.config.id.version(), tx_response.height.0)
                     .map_err(|_| Error::invalid_height_no_source())?;
                 // Check if the tx is valid
-                let tx_result = applied
-                    .read_attribute::<BatchAttr<'_>>()
-                    .expect("The batch attribute should exist");
-                let code = applied
-                    .read_attribute::<CodeAttr>()
-                    .expect("The code attribute should exist");
-                if code != ResultCode::Ok {
+                if tx_response.code != ResultCode::Ok {
                     return Ok(vec![IbcEventWithHeight::new(
                         IbcEvent::ChainError(format!(
-                            "The transaction was invalid: TxResult {tx_result}",
+                            "The transaction was invalid: TxResult {:#?}",
+                            tx_response.batch
```

Comment:
I'm not too familiar with Namada's TxResponse, but would adding `info` or `log` field give additional debug information which could be helpful or does `batch` contain everything?
https://github.com/anoma/namada/blob/c09d49fdfaa44be939424ff256082ef968b891f3/crates/sdk/src/rpc.rs#L695-L698